carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java

```diff
@@ -207,21 +209,26 @@ public void run() {
     }
 
     private void certificatesLifecycle() {
-        boolean updateDB = true;
-        boolean notifyCertAvailChanged = false;
-        for (String domain : certificates.keySet()) {
+        List<String> domains = new ArrayList(certificates.keySet()); // to get a predictable execution sequence in tests
+        Collections.sort(domains);
+        for (String domain : domains) {
+            boolean updateDB = true;
+            boolean notifyCertAvailChanged = false;
             try {
                 CertificateData cert = loadOrCreateDynamicCertificateForDomain(domain);
                 switch (DynamicCertificateState.valueOf(cert.getState())) {
                     case WAITING: { // certificate waiting to be issues/renew
                         LOG.info("Certificate ISSUING process for domain: " + domain + " STARTED.");
                         Order order = client.createOrderForDomain(domain);
                         cert.setPendingOrderLocation(order.getLocation());
+                        LOG.fine("Pending order location for domain " + domain + ": " + order.getLocation());
```

Comment:
you can leave at INFO, this is important and we are not ordering new certificate frequently